### PR TITLE
Buttons

### PR DESCRIPTION
**Pull request by wfortin**
New buttons following our guidelines:

### Buttons
* No more btn-alternate. 
* Default button, primary, secondary and danger modifiers
* Large modifier
* Prepend modifier

### Other changes
* Moved colors classes, fill- , bg- and text- to utility/colors.scss and included at the end since they could not override the button's colors.

screen shot 2015-11-10 at 3 53 05 pm

I'm implementing them in the auth server and the admin.

### CODE REVIEW

**Comment by pastjean** [comment]
:+1:

File: docs/_components/buttons/01_button.html [hunk 1]
```diff
@@ -6,3 +6,5 @@
 ---
 <a class="btn">Link</a>
 <button type="button" class="btn">Button</button>
+<button type="button" class="btn" disabled>Button disabled</button>
+<button type="button" class="btn state-disabled">Button disabled</button>
```
Comment (marilynechan) [inline docs/_components/buttons/01_button.html]: Shouldn't we only use the disabled attribute? Are there cases where we'll want the class but not the attribute?

File: docs/_components/buttons/02_button-colors.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (marilynechan) [inline docs/_components/buttons/02_button-colors.html]: Rename the file to button-modifiers?
Also, I'm not sure it's necessary to have 2 files for this. It could be in the `01_button.html` file.

**Comment by GermainBergeron** [comment]
:ship: :it:

File: docs/_components/buttons/03_button-sizes.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+type: buttons
+title: Button color modifiers
+usage: Fancy larger buttons? Use `.mod-large` for additional sizes.
```
Comment (marilynechan) [inline docs/_components/buttons/03_button-sizes.html]: Not sure about the comment - we only have one size modifier, and you say additional size*s*.

File: docs/_components/buttons/04_button-prepend.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        G+
+    </i>
+    Log in with Google
+</button>
+
+<button type="button" class="btn mod-prepend mod-large">
+    <i class="btn-prepend">
+        G+
+    </i>
+    Log in with Google
```
Comment (marilynechan) [inline docs/_components/buttons/04_button-prepend.html]: We should use something more generic here.

File: scss/components/application-container.scss [hunk 5]
```diff
@@ -1,7 +1,8 @@
 .application-container {
   height: calc(100vh - #{$header-height});
-  overflow-y: scroll;
```
Comment (marilynechan) [inline scss/components/application-container.scss]: Sorry... hope you guys don't have merge conflicts. Pierre-Olivier also made these changes in his branch...

File: docs/_components/buttons/01_button.html (hunk 1 above)
Comment (pastjean) [inline docs/_components/buttons/01_button.html]: :+1:

File: scss/elements/button-prepend.scss [hunk 6]
```diff
@@ -0,0 +1,33 @@
+.btn {
```
Comment (marilynechan) [inline scss/elements/button-prepend.scss]: Why are we naming the file button-prepend but the classes btn-prepend? I don't care which (but really, is this the only shortened word we use?), but stick to one naming please.

File: scss/elements/button-prepend.scss [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    padding-left: $button-height + $button-padding-x;
+
+    .btn-prepend {
+      position: absolute;
+      top: 0;
+      left: 0;
+
+      display: inline-block;
+      width: $button-height;
+      height: ($button-height - 1);
```
Comment (marilynechan) [inline scss/elements/button-prepend.scss]: In general, I don't like absolute positionning and I don't think it's necessary in this case.
Shouldn't the width & height simply adjust to the content of the button prepend?

File: scss/elements/button.scss [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+$button-line-height: 15px;
+$button-font-weight: 700;
+$button-padding-x: 16px;
+$button-padding-y: 8.5px;
+$button-border-width: 1px;
+$button-border-radius: 2px;
+$button-height: 34px;
+$button-large-height: 50px;
+
+@mixin button-variant($color, $background) {
```
Comment (marilynechan) [inline scss/elements/button.scss]: Nice!

File: scss/elements/button.scss [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    background-color: lighten($background, 20%);
+    border: $button-border-width solid lighten($background, 20%);
+
+    cursor: default;
+  }
+}
+
+
+.btn {
+  position: relative;
```
Comment (marilynechan) [inline scss/elements/button.scss]: Why is this needed?

File: docs/_components/buttons/01_button.html (hunk 1 above)
Comment (wfortin) [inline docs/_components/buttons/01_button.html]: I checked on bootstrap and basscss, and they all have a state-disabled class of some sort. I think it's because you sometimes want a <a class="btn">Link</a> and disabled is not available for those. I could update the sample to use the a tag for clarity. What do you think ?

File: scss/elements/button.scss [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+  vertical-align: middle;
+  white-space: nowrap;
+
+  background-color: transparent;
+  border: $button-border-width solid $medium-grey;
+  border-radius: $button-border-radius;
+
+  cursor: pointer;
+
+  &:hover, &:focus {
```
Comment (marilynechan) [inline scss/elements/button.scss]: Shouldn't you be using your button-variant mixin instead?

File: docs/_components/buttons/02_button-colors.html (hunk 2 above)
Comment (wfortin) [inline docs/_components/buttons/02_button-colors.html]: Yeah the file was originally button-modifiers but since there are colors and prepend and maybe others coming i prefered beign more precise.

File: docs/_components/buttons/01_button.html (hunk 1 above)
Comment (marilynechan) [inline docs/_components/buttons/01_button.html]: Good idea!

**Comment by marilynechan** [comment]
Where is the 'secondary' button used? I didn't notice that in the mockups.
Other that my few comments, :shipit: !

File: docs/_components/buttons/03_button-sizes.html (hunk 3 above)
Comment (wfortin) [inline docs/_components/buttons/03_button-sizes.html]: Yeah wasn't inspired so i copied bootstrap's description :stuck_out_tongue: , i'll find something better.

File: scss/elements/button-prepend.scss (hunk 6 above)
Comment (wfortin) [inline scss/elements/button-prepend.scss]: Good point, i'll rename to `btn-prepend`

File: scss/elements/button-prepend.scss (hunk 7 above)
Comment (wfortin) [inline scss/elements/button-prepend.scss]: I'll see what other options we have, i also don't like absolute positioning but i'm afraid of floats and align, maybe flex.

File: scss/elements/button.scss (hunk 9 above)
Comment (wfortin) [inline scss/elements/button.scss]: For the prepend, i tought i'd put it directly in the button, but if prepend is not absolute, i'll remove that.

**Comment by wfortin** [comment]
Secondary isn't used for the moment, i could remove it.

File: scss/elements/button.scss (hunk 10 above)
Comment (wfortin) [inline scss/elements/button.scss]: that could work, i'll check that.

File: docs/_components/buttons/03_button-sizes.html (hunk 3 above)
Comment (wfortin) [inline docs/_components/buttons/03_button-sizes.html]: Fixed.

File: scss/elements/button-prepend.scss (hunk 6 above)
Comment (wfortin) [inline scss/elements/button-prepend.scss]: done.

File: scss/elements/button.scss (hunk 9 above)
Comment (wfortin) [inline scss/elements/button.scss]: Fixed.

**Comment by wfortin** [comment]
Removed `.mod-secondary` as it is not used.
